### PR TITLE
[irrc-71] validations to status updates in backend

### LINKED ISSUE #44: Ensure consistency when making `statusUpdates`
Before accepting a `statusUpdate` from some `prevStatus` to `nextStatus`, the backend should check that the relevant student's current `statusID` indeed matches the claimed `prevStatus`.

### PR DESCRIPTION
**Pull request by lauratsui**
added validations for status updates - previous status cannot be the same as next status, and prev status should be the same as current status in database. Fixes #44

### CODE REVIEW

**Review by therealyingtong — APPROVED** [review]

File: backend/src/routes/statusUpdates.js [hunk 1]
```diff
@@ -15,6 +16,30 @@ router.post('/', async (req, res) => {
   delete req.body.PrevStatusString
   delete req.body.NextStatusString
 
+  if (prevStatus.StatusID == nextStatus.StatusID) {
```
Comment (therealyingtong) [inline backend/src/routes/statusUpdates.js]:
```suggestion
  if (req.body.PreviousStatusID == req.body.NextStatusID) {
```

File: backend/src/routes/statusUpdates.js [hunk 2]
```diff
@@ -15,6 +16,30 @@ router.post('/', async (req, res) => {
...
+      data: {
+        previousStatus: prevStatus.StatusID,
+        nextStatus: nextStatus.StatusID
+      }
+    })
+  }
+
+  const student = await students.getStudentById(req.body.StudentID)
+
+  if (student.StatusID !== prevStatus.StatusID) {
```
Comment (therealyingtong) [inline backend/src/routes/statusUpdates.js]:
```suggestion
  if (student.StatusID !== req.body.PreviousStatusID) {
```

**Review by therealyingtong — CHANGES_REQUESTED** [review]

File: backend/src/routes/statusUpdates.js [hunk 3]
```diff
@@ -15,6 +16,30 @@ router.post('/', async (req, res) => {
   delete req.body.PrevStatusString
   delete req.body.NextStatusString
 
+  if (prevStatus.StatusID == nextStatus.StatusID) {
+    return res.status(422).send({
+      message: 'Previous Status and Next Status should not be the same.',
+      type: 'StatusError',
+      data: {
+        previousStatus: prevStatus.StatusID,
+        nextStatus: nextStatus.StatusID
```
Comment (therealyingtong) [inline backend/src/routes/statusUpdates.js]:
```suggestion
        previousStatus: req.body.PreviousStatusID,
        nextStatus: req.body.NextStatusID
```

File: backend/src/routes/statusUpdates.js [hunk 4]
```diff
@@ -15,6 +16,30 @@ router.post('/', async (req, res) => {
...
+  }
+
+  const student = await students.getStudentById(req.body.StudentID)
+
+  if (student.StatusID !== prevStatus.StatusID) {
+    return res.status(409).send({
+      message: 'Previous status must match current student status.',
+      type: 'StatusMismatch',
+      data: {
+        submittedStatus: prevStatus.StatusID,
```
Comment (therealyingtong) [inline backend/src/routes/statusUpdates.js]:
```suggestion
        submittedStatus: req.body.PreviousStatusID,
```

**Review by therealyingtong — CHANGES_REQUESTED** [review]

**Comment by thchia** [comment]
@therealyingtong suggested changes made!

**Review by therealyingtong — APPROVED** [review]

**Review by xuanjane — APPROVED** [review]
LGTM :)

**Pull request merged**